torchtext/csrc/sentencepiece.cpp

```diff
@@ -0,0 +1,89 @@
+#include <sentencepiece_trainer.h>
+#include <torch/script.h>
+
+namespace torchtext {
+namespace {
+
+struct SentencePiece : torch::CustomClassHolder {
+private:
+  sentencepiece::SentencePieceProcessor processor_;
+
+public:
+  std::string content_;
+  explicit SentencePiece(const std::string &content) : content_(content) {
+    const auto status = processor_.LoadFromSerializedProto(content_);
+    if (!status.ok()) {
+      throw std::runtime_error("Failed to load SentencePiece model. Error: " +
+                               status.ToString());
+    }
+  }
+
+  std::vector<std::string> Encode(const std::string &input) const {
```

Comment:
If you want, you could play with iterables here. You pass an object that yields strings and return and object that yields iterators over strings. That could be quite powerful when it comes to avoiding the overhead of writing out a lot of text while processing.

Reply:
Just a reminder. Iterator is not supported by torchscript but we expect those transforms jitable.